common/djangoapps/track/views/segmentio.py

```diff
@@ -129,20 +127,6 @@ def track_segmentio_event(request):  # pylint: disable=too-many-statements
     # Build up the event context by parsing fields out of the event received from Segment
     context = {}
 
-    library_name = segment_context.get('library', {}).get('name')
-    source_map = getattr(settings, 'TRACKING_SEGMENTIO_SOURCE_MAP', {})
-    event_source = source_map.get(library_name)
-    if not event_source:
-        raise EventValidationError(WARNING_IGNORED_SOURCE)
-    else:
-        context['event_source'] = event_source
-
-    if 'name' not in segment_properties:
-        raise EventValidationError(ERROR_MISSING_NAME)
-
-    if 'data' not in segment_properties:
-        raise EventValidationError(ERROR_MISSING_DATA)
-
     # Ignore event types and names that are unsupported
     segment_event_type = full_segment_event.get('type')
     segment_event_name = segment_properties['name']
```

Comment:
This will barf if name is not present.  So put it after the check for name on L152.